### PR TITLE
Generate smaller bundle options with limited timezone data

### LINKED ISSUE #67: wishlist: offer options for smaller data bundles
Because of moment/moment-timezone#697, the timezone data upon which `js-joda-timezone` relies has grown massively in size recently, and therefore so has `js-joda-timezone` itself.

| version | dist/js-joda-timezone.min.js | gzip -9 |
| --- | --- | --- |
| 2.0.1 | 186973 bytes | 25331 bytes |
| 2.0.2 | 931347 bytes | 36084 bytes |

`moment-timezone` is [now offering](https://github.com/moment/moment-timezone/tree/develop/builds) builds of the package with some of the data removed if it is not relevant for timestamps within ±5 years or so of the present. It would be good if `js-joda-timzeone` could copy this and offer smaller builds of the package for use cases where this makes sense.

### PR DESCRIPTION
**Pull request by JKillian**
Fixes #67.

## Changes

This PR changes the build process such that three new dev/prod bundles are produced:

* `js-joda-timezone-10-year-range.js` which includes timezone data for +- five years from the current version's release
* `js-joda-timezone-1970-2030.js` which includes timezone data for covers from 1970 to 2030
* `js-joda-timezone-2012-2022.js` which includes timezone data for covers from 2012 to 2022

These bundles intentionally [echo what moment-timezone provides](https://github.com/moment/moment-timezone/blob/c377ac8fd7ff6c2fc52e2f432034ac5727dba1d8/Gruntfile.js#L15L20).

Users can choose to use these bundles in their code by simply importing them:

```ts
import 'js-joda-timezone/dist/js-joda-timezone-1970-2030'
```
Here are the file sizes of the new bundles compared to the current bundle:

**Minified bundle sizes**:

bundle | size
-- | --
all | 911 KB
10-year-range | 34 KB
1970 - 2030 | 150 KB
2012 - 2022 | 34 KB

## Build Steps

* Copy over latest data from moment-timezone:
  - moment-timezone's _unpacked_ `latest.json` should be copied over to `data/unpacked/latest.json`.
  - moment-timezone's dated packed data, such as `2018h.json`, can be copied over to `data/packed/<name>` if it isn't yet.
* Run `npm run transform-data` to generate all the `data/packed/latest*.json` files. (Note that CI doesn't have to run this step since I checked the generated files in.)
* Run `npm run build-dist` to build all the `.js` bundles based on the `latest*.json` files available.

## A few caveats / questions for maintainers:

* `npm` was acting strangely for me and modifying much too mock of the `package-lock.json` file so I had to manually `add -p` it and it's possible things went wrong. Feel free to change it or make any other changes to this PR also, or comment and I'm happy to make changes.
* Do we need to change tests at all to account for these new builds?
* How does js-joda react if it doesn't have timezone data available? For example, if you're using `js-joda-timezone-10-year-range.js` and you have a `ZonedDateTime` outside of that range, is the behavior graceful?
* I added `data/packed/2018h.json` through `data/packed/2019a.json` while I was modifying things, but happy to remove them if that's too orthogonal to this PR.

### CODE REVIEW

File: transform-data.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ */
+const FILES_TO_WRITE = {
+    '': [0, 9999],
+    '-1970-2030': [1970, 2030],
+    '-2012-2022': [2012, 2022],
+    '-10-year-range': [currentYear-5, currentYear+5],
+};
+
+const writeData = (suffix, startYear = 0, endYear = 9999) => {
+    const packedData = moment.tz.filterLinkPack(tzdbLatest, startYear, endYear);
```
Comment (JKillian) [inline transform-data.js]: note: this is the same way that `moment-timezone` produces its smaller bundles.

**Comment by mlc** [comment]
i'm no expert in the relevant details, but this looks exactly like what I was wanting :smile_cat:

**Comment by pithu** [comment]
Hello @JKillian,

thanks a lot for using js-joda and contributing.  Your PR is looking great, i probably would have done it very similar. What would be nice to have, is a `Howto` build timezone data, as describe by you in the `Build Steps`.

To your questions:
> npm was acting strangely for me and modifying much too mock of the package-lock.json ...
The differences in the `package-lock.json` are probably because we are using different node nvm version. It should be ok to have some small difference.

> Do we need to change tests at all to account for these new builds?
Having karma tests for all build version in dist would be also nice to have, but the test need to be adapted for that, because some contain tests for older tzdb entries.

> How does js-joda react if it doesn't have timezone data available? 
I did not test it yet, but it should be no problem, i expect that `js-joda` will use the earliest available tzdb data entry. So the result might be incorrect, but calculation should not fail.

**Comment by JKillian** [comment]
Amazing! Thanks for the merge @pithu. Let me know when you release and we'll pick up the new version in our codebase.

> Having karma tests for all build version in dist would be also nice to have, but the test need to be adapted for that, because some contain tests for older tzdb entries.

Agreed! For a later PR perhaps.

> I did not test it yet, but it should be no problem, i expect that js-joda will use the earliest available tzdb data entry. So the result might be incorrect, but calculation should not fail.

I think this is the correct behavior, sounds good!

---

Also, I'd love to be able to make some space-savings over in [js-joda-locale also](https://github.com/js-joda/js-joda-locale/issues/124)! Let me know over in that thread if you have any ideas for improvements.

**Pull request merged**